tests/dem/normal_force.cc

```diff
@@ -102,24 +103,51 @@ test()
     num_particles);
   PWNonLinearForce<dim>         pwcf1;
   VelocityVerletIntegrator<dim> Integ1;
+  double                        distance;
 
   for (double time = 0; time < 0.00115; time += dt)
     {
       auto particle = particle_handler.begin();
       particle->get_properties()[DEM::PropertiesIndex::force_x] = 0;
       particle->get_properties()[DEM::PropertiesIndex::force_y] = 0;
-      particle->get_properties()[DEM::PropertiesIndex::force_z] = 0;
-
-      pw2.pw_Fine_Search(pwContactList, pwContactInfo, dt);
-      auto pw_pairs_in_contact_iterator = pwContactInfo.begin();
-      auto pw_contact_information_iterator =
-        pw_pairs_in_contact_iterator->begin();
-
-      pwcf1.calculate_pw_contact_force(&pwContactInfo, dem_parameters);
+      if (dim == 3)
+        {
+          particle->get_properties()[DEM::PropertiesIndex::force_z] = 0;
+        }
 
-      Integ1.integrate(particle_handler, g, dt);
-      if (pw_contact_information_iterator->second.normal_overlap > 0)
+      distance = length + particle->get_location()[0] -
+                 particle->get_properties()[DEM::PropertiesIndex::dp] / 2.0;
+      if (distance > 0.0)
         {
+          Integ1.integrate(particle_handler, g, dt);
+        }
+      else
```

Comment:
So the root of the cause was that you were trying to access this information when there was no contact I guess right?

Reply:
Exactly. I defined a variable distance in the test (which is equal to the normal overlap), and if the distance is positive, all the required functions for force calculations are called.

Reply:
Seems good to me :)!